client/my-sites/sharing/buttons/buttons.jsx

```diff
@@ -41,12 +45,31 @@ class SharingButtons extends Component {
 	};
 
 	saveChanges = event => {
+		const {
+			isJetpack,
+			isLikesModuleActive,
+			siteId
+		} = this.props;
+
 		event.preventDefault();
+
 		this.props.saveSiteSettings( this.props.siteId, this.state.values );
 		if ( this.state.buttonsPendingSave ) {
 			this.props.saveSharingButtons( this.props.siteId, this.state.buttonsPendingSave );
 		}
 		this.props.recordGoogleEvent( 'Sharing', 'Clicked Save Changes Button' );
+
+		if ( ! isJetpack || isLikesModuleActive !== false ) {
+			return;
+		}
+
+		const updatedSettings = this.getUpdatedSettings();
+		if ( updatedSettings.disabled_likes ) {
+			return;
+		}
+
+		this.props.activateModule( siteId, 'likes', true )
+			.then( () => this.props.fetchModuleList( siteId ) );
```

Comment:
We can use the [`<QueryJetpackModules />`](https://github.com/Automattic/wp-calypso/tree/master/client/components/data/query-jetpack-modules) query component to fetch this data?